dbt/adapters/spark/impl.py

```diff
@@ -58,24 +58,29 @@ def get_relation_type(self, relation, model_name=None):
     def list_relations_without_caching(self, information_schema, schema,
                                        model_name=None):
         kwargs = {'information_schema': information_schema, 'schema': schema}
-        results = self.execute_macro(
-            LIST_RELATIONS_MACRO_NAME,
-            kwargs=kwargs,
-            release=True
-        )
+        try:
+            results = self.execute_macro(
+                LIST_RELATIONS_MACRO_NAME,
+                kwargs=kwargs,
+                release=True
+            )
+        except dbt.exceptions.RuntimeException as e:
+            if hasattr(e, 'msg') and f"Database '{schema}' not found" in e.msg:
+                return []
 
         relations = []
         quote_policy = {
             'schema': True,
             'identifier': True
         }
-        for _database, name, _ in results:
+        for _database, name, _, information in results:
+            rel_type = ('view' if 'Type: VIEW' in information else 'table')
```

Comment:
we've had issues in the past with other adapters where we assumed that a relation was either of type "table" or "view". In practice, some databases support other relation types, like materialized views or external tables. In this case, I think this logic looks A-OK, but I did want to flag it as something to keep in mind for the future!